File: leetcode-algorithms/033. Search in Rotated Sorted Array/solution.py
def search(nums, target):
	"""
	:type nums: List[int]
	:type target: int
	:rtype: int
	"""
	
	lo = 0
	hi = len(nums)
	while lo < hi:
		mid = (hi+lo) // 2
		num = 0
		if (nums[mid]<nums[0]) == (target < nums[0]):
			num = nums[mid]
		else:
			if target < nums[0]:
				num = -float('inf')
			else:
				num = float('inf')

		if target > num:
			lo = mid + 1
		elif target < num:
			hi = mid
		else:
			return mid
	return -1



if __name__ == '__main__':
	nums = [4,5,6,7,0,1,2]
	res = search(nums,1)
	print(res)

File: leetcode-algorithms/033. Search in Rotated Sorted Array/solution.cpp
#include <iostream>
#include <vector>
#include <cmath>

using namespace std;

int search(vector<int>& nums, int target) {
	int lo = 0, hi = nums.size();

	while(lo<hi){
		int mid = (lo+hi) /2;
		double num = (nums[mid]<nums[0]) == (target<nums[0])
					? nums[mid]
					: target < nums[0]?-INFINITY:INFINITY;
		if(num<target)
			lo = mid + 1;
		else if(num>target)
			hi = mid;
		else
			return mid;
	}
	return -1;
}

int main()
{
	vector<int> nums{4,5,6,7,0,1,2};
	int res = search(nums,2);
	cout<<res<<endl;
}